src/protocol/context/malicious.rs

```diff
@@ -3,6 +3,7 @@ use std::marker::PhantomData;
 
 use async_trait::async_trait;
 use futures::future::{try_join, try_join_all};
+use once_cell::sync::OnceCell;
```

Comment:
I don't see you using `OnceCell` here, which is good, I think.

Reply:
Only in `PhantomData`, for now. I got aggressive about putting placeholders in all the types to make sure that I rooted out anything that was going to have a problem with invariance over the lifetime parameter. In cases where we think we may want to make use of that in the foreseeable future, then keeping the placeholder avoids regressions. But I don't see a specific need for it in the malicious context structs, so I'm fine dropping it. (At the moment, the malicious contexts contain a semi-honest context, so they'll inherit the invariance that way.)